arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/CancellableF.kt

```diff
@@ -28,7 +28,7 @@ class CancellableF : ArrowFxSpec(spec = {
         latch.complete(Result.success(Unit))
         CancelToken { cancelled.complete(true) }
       }
-    }.startCoroutineCancellable(CancellableContinuation { })
+    }.startCoroutineCancellable(CancellableContinuation(coroutineContext) { })
```

Comment:
This deadlocks since [`coroutineContext` is a single threaded context](https://github.com/kotest/kotest/blob/cf7a5ed39361cef281d6494205091db7fc7ff491/kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/KotestEngine.kt#L132) given the default configuration from Kotest is = 1, and `c.invoke()` is performing `unsafeRunSync` (with lock) to perform a back-pressured cancellation.

This can be fixed by `ForkConnected { c.invoke() }` but we should perhaps also evaluate if we want to keep this cancellation model. We should perhaps evaluate if that is the only way we want to expose cancellation, it would probably be safer to expose both `suspend fun cancel(): Unit` and `fun unsafeCancel(): Unit` where `unsafeCancel(): Unit` doesn't use `unsafeRunSync` behind the scenes but rather `unsafeRunAsync` which would remove the requirement of a lock and would not make this deadlock.

Reply:
This would also fixes it, which keeps the same semantics but suspends instead of locks.

```kotlin
val cancelled = Promise<Boolean>()
val latch = UnsafePromise<Unit>()
val conn = SuspendConnection()
val c = suspend {
  cancellable<Unit> {
    latch.complete(Result.success(Unit))
    CancelToken { cancelled.complete(true) }
  }
}.startCoroutineCancellable(CancellableContinuation(coroutineContext, conn) { })
latch.join()
conn.cancel()
cancelled.get() shouldBe true
```

Reply:
Hmm I have `cancelableF executes generated task uninterruptedly` failing locally as well. I don't think I understand the tests well enough to fix them just yet. I'll look at this when I have some more time later this week^^